### ISSUE TITLE
YTC doesn't work in Firefox 38.0.5

### ISSUE THREAD

**Issue by Atulin** [YePpHa/YouTubeCenter#1885]
The options do not show. It is not a conflict with other addons too - I tried running FF with only YTC enabled, I got the same result.

What's weird, is that even the GreaseMonkey YTC scrip doesn't work at all...

**Comment by Yonezpt** [comment]
If you are not using the developer version the download it from here: https://github.com/YePpHa/YouTubeCenter/wiki/Developer-Version

**Comment by mighelch** [comment]
It doesn't work at all. It's not problem with ff. It has stopped working 2 or 3 weeks ago for me.

**Comment by Yonezpt** [comment]
@mighelch The latest developer version is still working for me with Firefox, are you using dev version 525?

**Comment by mighelch** [comment]
Yes. The 525

**Comment by mighelch** [comment]
Ok. I have disabled all GM scripts > reenabled them all and now YTC works. Some features of ytc don't work though.

**Comment by Yonezpt** [comment]
Those are probably features that need to be updated by the author, but the rest is still working as you could confirm by the steps you took.

**Comment by mighelch** [comment]
Yes. They're already reported in issues.

**Comment by Atulin** [comment]
Using developer version works like a charm for me as well. Thank you.

**Comment by ErdeFB** [comment]
I am also having the same issue on Chromium. After compiling a new version of the browser, the add-on does not load most of the time.

I figured out a temporary workaround which is to go to extensions list and scroll to YTC and click options. It opens a new tab with Youtube and opens the YTC settings menu and as long as I remain on that tab and open new videos or video tabs from it, the add-on stays enabled.

**Comment by lunghd** [comment]
I am also having this issue with 38.0.5; -zero- YTC functionality. There is no access to gui and neither the FireFox extension nor the user script will load now. It has been misbehaving for a few weeks with a prior version of FireFox but was, for the most part, still functional.

I have tried both the current and developer versions in both the FireFox extension and the user script and also disabled & enabled, refreshed, cleared cookies etc - no joy. 

A separate issue would be to move the GUI to about:config instead of hoping YouTube won't hide that itty bitty "X" to access the GUI (as they did previously). 

YouTube keeps juggling things around and likely is doing all it can to prevent scripts from working. (The asshats at YT also boned my dvd player's layout making it nearly impossible to navigate.)

**Comment by Yonezpt** [comment]
@lunghd The latest developer version 526 is loading fine for me on the same Firefox version. Make sure to disable any other add-on or userscript except YTC, it is possible that one of them might be causing conflict. Another possibility is that you might have a YTC bugged setting which is preventing YTC to function properly, in which case you could try deleting the settings file which are located here:

 - For the add-on version: `C:\Users\M3o\AppData\Roaming\Mozilla\Firefox\Profiles\ <YOUR_PROFILE> \YouTubeCenterStorage` delete the `YouTubeCenterSettings.data` file

 - For the userscript version: `C:\Users\M3o\AppData\Roaming\Mozilla\Firefox\Profiles\ <YOUR_PROFILE> \YouTubeCenterStorage` delete any file that starts with `YouTube_Center`

**Comment by lunghd** [comment]
Took a while to find those - I have Win7 64 & it appears to stuff things elsewhere.

C:\Users\<username>\AppData\Roaming\Mozilla\Firefox\Profiles\<alphanumeric_random_gibberish>\YouTubeCenterStorage

C:\Users\<username>\AppData\Roaming\Mozilla\Firefox\Profiles\<alphanumeric_random_gibberish>\scriptish_scripts\youtubecenter

Renamed old files, uninstalled script, removed extension, restarted browser, reinstalled Dev. v526 and some of the features appear to function however I still can't access the user interface / GUI. This appears to be happening with several other YouTube scripts as well... likely an active effort by YT.

**Comment by Yonezpt** [comment]
Did you also disabled any other add-on/userscript except YTC to rule out the possibility of a conflict? Especially adblockers, there have been some cases when YTC UI was being blocked by an adblocker.

**Comment by lunghd** [comment]
Yep, although I still had some extensions running.

Went back and disabled -ALL- scripts and -ALL- extensions except for YTC - no change except for losing my grid view, which is controlled by the YouTube Subscriptions Grid extension. (This appears to be the only fully functional extension right now.)

I have AdBlock Plus with YouTube whitelisted but went back, disabled it entirely, restarted browser & no change. 

I also have a hosts file - YouTube tries to open an advert but that iframe fails to connect or present an advert. Never even saw that iframe when YTC was working.

I may roll back to an earlier version of FireFox to try that but something tells me that YouTube is actively trying to prevent us from using scripts etc.

**Comment by Yonezpt** [comment]
It's very strange because I am also using Win7 x64 + FF 38.0.5 + YTC 526 and the UI works, the little gear near the notifications bell shows correctly and clicking on it opens its settings.

The only thing left that I can think of to try is to test-run Firefox with a new profile with just YTC installed, no other add-on/userscript. Create a new profile: https://support.mozilla.org/en-US/kb/profile-manager-create-and-remove-firefox-profiles#w_starting-the-profile-manager and install just YTC developer version 526

**Comment by lunghd** [comment]
Thanks again but still no joy. Created new profile, downloaded scriptish & YTC, went to YouTube and it wasn't working. 
I even tried disabling my Hosts file... that was the only thing left although I can't imagine how it would affect things... and it didn't; no change.

Ok - so it works for you & some other folks... while others report this, or similar issues. I can only assume it's something to do with how YouTube has changed things combined with how some of our systems are set up. (Again - no idea what or how etc...)

I appreciate the input & will keep tabs on this thread. If I can figure out whatthehell is going on I'll post back as well.

**Comment by lunghd** [comment]
I wanted to post back here since you were helpful in eliminating possibilities. I think there is a 'perfect storm' combination going on with FireFox, Scriptish & who knows what else... 

I uninstalled Firefox 38.0.5 & installed the previous version 37.0.2 . 

No change with that but I had also stumbled upon thread #1563 which mentioned an issue with Scriptish that caught my attention as it mentioned that it had been deprecated.

I tried to uninstall Scriptish & use Greasemonkey & found that the uninstall process froze.. and then a couple of other scripts & extensions also froze as if unable to install or uninstall (depending on which one it was). Hmmm - 

I took a screenshot of all extensions & add-ons so I could recover them & used the Mozilla 'Refresh' option which reset FireFox to it's basic settings (while saving passwords, bookmarks, tabs etc).

I reinstalled Greasemonkey & YTC and Voila! It's back doing it's wonderful YTC thang on YouTube!

I'm not sure what exactly was the root cause - it appears that Scriptish was a part of the problem while FireFox was hanging up on installs and uninstalls of add-ons & scripts.

Thanks again for your help - it helped immensely! Whoo hoo!

**Comment by Yonezpt** [comment]
@lunghd Thanks for posting back, that information will be very useful for others that might find themselves in the same situation as you.
